### PR TITLE
Feedly unit tests and fixes

### PR DESCRIPTION
**Pull request by kielgillard**
- [x] Marking articles read/unread for Feedly is broken. May have been prior to this PR. #1042
- [ ] Feedly must be manually refreshed until I can figure out how other accounts automatically refresh at start up.

### CODE REVIEW

File: Frameworks/Account/AccountTests/TestAccountManager.swift [hunk 1]
```diff
@@ -23,16 +23,17 @@ class TestAccountManager {
 	func createAccount(type: AccountType, username: String? = nil, password: String? = nil, transport: Transport) -> Account {
 		
 		let accountID = UUID().uuidString
-		let accountFolder = accountsFolder.appendingPathComponent("\(type.rawValue)_\(accountID)").absoluteString
+		let accountFolder = accountsFolder.appendingPathComponent("\(type.rawValue)_\(accountID)")
 		
 		do {
-			try FileManager.default.createDirectory(atPath: accountFolder, withIntermediateDirectories: true, attributes: nil)
+			try FileManager.default.createDirectory(at: accountFolder, withIntermediateDirectories: true, attributes: nil)
+//			try FileManager.default.createDirectory(atPath: accountFolder, withIntermediateDirectories: true, attributes: nil)
```
Comment (kielgillard) [inline Frameworks/Account/AccountTests/TestAccountManager.swift]: I'll need to revert this.

File: Frameworks/Account/Feedly/FeedlyAccountDelegate.swift [hunk 2]
```diff
@@ -243,9 +244,9 @@ final class FeedlyAccountDelegate: AccountDelegate {
 										  log: log)
 		
 		//TODO: Figure out how other accounts get refreshed automatically.
-		refreshAll(for: account) { result in
-			print("sync after initialise did complete")
-		}
+//		refreshAll(for: account) { result in
+//			print("sync after initialise did complete")
+//		}
```
Comment (kielgillard) [inline Frameworks/Account/Feedly/FeedlyAccountDelegate.swift]: This disables the automatic syncing at start up.

File: Frameworks/Account/Folder.swift [hunk 3]
```diff
@@ -102,6 +102,14 @@ public final class Folder: DisplayNameProvider, Renamable, Container, UnreadCoun
...
 		postChildrenDidChangeNotification()
 	}
+	
+	public func removeFeeds(_ feeds: Set<Feed>) {
+		guard !feeds.isEmpty else {
+			return
+		}
+		topLevelFeeds.subtract(feeds)
+		postChildrenDidChangeNotification()
+	}
```
Comment (kielgillard) [inline Frameworks/Account/Folder.swift]: @vincode-io is this change okay?

File: Frameworks/Account/AccountTests/TestAccountManager.swift (hunk 1 above)
Comment (kielgillard) [reply Frameworks/Account/AccountTests/TestAccountManager.swift]: If I revert this, SQL barfs in the Console with messages such as:
```
2019-10-03 17:45:10.029809+1000 xctest[18281:9334018] DB Path: file:/var/folders/2_/b1_slhkn0yx4nmk268xbtl6m0000gn/T/17_E0E1B831-21A7-4641-96A3-A57EA9512CD1/Sync.sqlite3
2019-10-03 17:45:10.029982+1000 xctest[18281:9334341] DB Path: file:/var/folders/2_/b1_slhkn0yx4nmk268xbtl6m0000gn/T/17_E0E1B831-21A7-4641-96A3-A57EA9512CD1/DB.sqlite3
2019-10-03 17:45:10.030143+1000 xctest[18281:9334018] [logging] API call with unopened database connection pointer
2019-10-03 17:45:10.030168+1000 xctest[18281:9334341] [logging] API call with unopened database connection pointer
2019-10-03 17:45:10.030196+1000 xctest[18281:9334018] [logging] misuse at line 127840 of [95fbac39ba]
2019-10-03 17:45:10.030330+1000 xctest[18281:9334018] DB Error: 14 "unable to open database file"
2019-10-03 17:45:10.030396+1000 xctest[18281:9334341] [logging] misuse at line 127840 of [95fbac39ba]
2019-10-03 17:45:10.030494+1000 xctest[18281:9334018] DB Query: CREATE TABLE if not EXISTS syncStatus (articleID TEXT NOT NULL, key TEXT NOT NULL, flag BOOL NOT NULL DEFAULT 0, selected BOOL NOT NULL DEFAULT 0, PRIMARY KEY (articleID, key));
2019-10-03 17:45:10.030551+1000 xctest[18281:9334341] DB Error: 14 "unable to open database file"
2019-10-03 17:45:10.031228+1000 xctest[18281:9334018] DB Path: file:/var/folders/2_/b1_slhkn0yx4nmk268xbtl6m0000gn/T/17_E0E1B831-21A7-4641-96A3-A57EA9512CD1/Sync.sqlite3
2019-10-03 17:45:10.031228+1000 xctest[18281:9334341] DB Query: CREATE TABLE if not EXISTS articles (articleID TEXT NOT NULL PRIMARY KEY, feedID TEXT NOT NULL, uniqueID TEXT NOT NULL, title TEXT, contentHTML TEXT, contentText TEXT, url TEXT, externalURL TEXT, summary TEXT, imageURL TEXT, bannerImageURL TEXT, datePublished DATE, dateModified DATE, searchRowID INTEGER);
2019-10-03 17:45:10.031304+1000 xctest[18281:9334341] DB Path: file:/var/folders/2_/b1_slhkn0yx4nmk268xbtl6m0000gn/T/17_E0E1B831-21A7-4641-96A3-A57EA9512CD1/DB.sqlite3
2019-10-03 17:45:10.031377+1000 xctest[18281:9334341] [logging] API call with unopened database connection pointer
2019-10-03 17:45:10.031866+1000 xctest[18281:9334341] [logging] misuse at line 127840 of [95fbac39ba]
2019-10-03 17:45:10.031936+1000 xctest[18281:9334341] DB Error: 14 "unable to open database file"
2019-10-03 17:45:10.031998+1000 xctest[18281:9334341] DB Query: CREATE TABLE if not EXISTS statuses (articleID TEXT NOT NULL PRIMARY KEY, read BOOL NOT NULL DEFAULT 0, starred BOOL NOT NULL DEFAULT 0, userDeleted BOOL NOT NULL DEFAULT 0, dateArrived DATE NOT NULL DEFAULT 0);
2019-10-03 17:45:10.032057+1000 xctest[18281:9334341] DB Path: file:/var/folders/2_/b1_slhkn0yx4nmk268xbtl6m0000gn/T/17_E0E1B831-21A7-4641-96A3-A57EA9512CD1/DB.sqlite3
```

File: Frameworks/Account/Feedly/FeedlyAccountDelegate.swift (hunk 2 above)
Comment (kielgillard) [reply Frameworks/Account/Feedly/FeedlyAccountDelegate.swift]: I'll need to deal with this in another PR.

File: Frameworks/Account/Folder.swift (hunk 3 above)
Comment (vincode-io) [reply Frameworks/Account/Folder.swift]: Fine by me.

**Pull request merged**